am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java

```diff
@@ -1,42 +1,50 @@
 package uk.gov.hmcts.reform.amlib.helpers;
 
+import com.fasterxml.jackson.core.JsonPointer;
 import uk.gov.hmcts.reform.amlib.enums.Permission;
+import uk.gov.hmcts.reform.amlib.models.ExplicitAccessGrant;
 import uk.gov.hmcts.reform.amlib.models.ExplicitAccessMetadata;
-import uk.gov.hmcts.reform.amlib.models.ExplicitAccessRecord;
 
+import java.util.Map;
 import java.util.Set;
 
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ACCESSOR_ID;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ACCESS_TYPE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_TYPE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SECURITY_CLASSIFICATION;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SERVICE_NAME;
+
 public final class TestDataFactory {
 
     private TestDataFactory() {
         //NO-OP
     }
 
-    public static ExplicitAccessRecord createRecord(String resourceId,
-                                                    String accessorId,
-                                                    Set<Permission> explicitPermissions) {
-        return ExplicitAccessRecord.builder()
+    public static ExplicitAccessGrant createRecord(String resourceId,
```

Comment:
I would rename that factory method. Record in my mind set should only be used for DB records mapped to `ExplicitAccessRecord`.